### ISSUE TITLE
SourceContext  not outputing correctly in Serilog logger

### ISSUE THREAD

**Issue by madhon** [damianh/LibLog#40]
When logging using Seirlog when outputting the $sourcecontext property in the output template its not replacingith correctly with the context

serilog is configured as per

```
        Log.Logger = New LoggerConfiguration() _
            .Enrich.With(New ThreadIdEnricher()) _
            .Enrich.FromLogContext _
            .Enrich.WithMachineName _
            .Enrich.WithProcessId _
            .MinimumLevel.Is(Events.LogEventLevel.Debug) _
            .WriteTo.ColoredConsole(outputTemplate:=logLayout) _
            .WriteTo.File(logFileName, outputTemplate:=logLayout) _
            .CreateLogger()
```

outputTemplate is
```
{Timestamp:HH:mm} [{Level}] ({ThreadId}) ({SourceContext})  {Message}{NewLine}{Exception}
```

the context should be there eg.
```
11:01 [Information] (8) (AIMIE.MaintenanceService.Program)  OnStart at 03/05/2015 11:01:37
```

**Comment by damianh** [comment]
Smells like a duplicate of #25 (in essence) ?

Am open to suggestions here.

**Comment by damianh** [comment]
Ping @nblumhardt , any suggestions on what I can do here (if there is anything I can do) ? Cheers

**Comment by nblumhardt** [comment]
Howdy all! Not a duplicate of #25 - Serilog doesn't grab this dynamically from the call stack at all. To get this property in there, you need to get a type-specific sub-logger along these lines:

```csharp
var typeSpecific = someOtherLogger.ForContext("SourceContext", typeName)
```

Where I believe `typeName` in the default Serilog implementation is just `Type.Name` (IIRC).

Not sure if LibLog has any notion of this already, but it's pretty similar to log4net's `LogManager.GetLogger(Type)` mechanism I think.

HTH!

**Comment by damianh** [comment]
Thanks @nblumhardt ! 

Will see if I can hook this in.

**Comment by adamralph** [comment]
Is this fixed by #99?

**Comment by damianh** [comment]
Yes it appears so.
